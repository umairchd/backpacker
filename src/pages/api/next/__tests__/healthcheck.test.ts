/**
 * @jest-environment node
 */

import path from "node:path";

import handler from "../healthcheck.page";

import { createMocks } from "node-mocks-http";

const TEST_NAME = "healthcheck";

import nock from "nock";

describe(TEST_NAME, () => {
  let nockBack: nock.Back;

  beforeAll(async () => {
    nockBack = nock.back;
    nockBack.setMode(nockBack.currentMode);
    nockBack.fixtures = path.resolve(
      __dirname,
      `./__nock-fixtures__/${TEST_NAME}`
    );
  });

  beforeEach(() => {
    nock.enableNetConnect(/(localhost|127\.0\.0\.1)/);
  });

  afterEach(() => {
    nock.restore();
    nock.cleanAll();
    jest.resetAllMocks();
  });

  it("happy case", async () => {
    const { nockDone } = await nockBack("happy_case.json");

    const { req, res } = createMocks({
      method: "GET",
    });

    await handler(req as any, res as any);

    expect(res._getStatusCode()).toBe(200);
    expect(res._getData()).toMatchSnapshot();

    nockDone();
  });

  it("happy case post", async () => {
    const { nockDone } = await nockBack("happy_case.json");

    const { req, res } = createMocks({
      method: "POST",
    });

    await handler(req as any, res as any);

    expect(res._getStatusCode()).toBe(200);
    expect(res._getData()).toMatchSnapshot();

    nockDone();
  });
});
