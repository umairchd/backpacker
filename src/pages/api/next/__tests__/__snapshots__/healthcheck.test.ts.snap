// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`healthcheck happy case 1`] = `"{"status":"UP"}"`;

exports[`healthcheck happy case post 1`] = `"{"status":"UP"}"`;
